crates/pallets/starknet/src/lib.rs

```diff
@@ -747,14 +747,21 @@ pub mod pallet {
                 .longevity(T::TransactionLongevity::get())
                 .propagate(true);
 
-            // Make sure txs from same account are executed in correct order (nonce based ordering)
-            if let TxPriorityInfo::RegularTxs { sender_address, transaction_nonce, sender_nonce } = tx_priority_info {
-                valid_transaction_builder =
-                    valid_transaction_builder.and_provides((sender_address, Felt252Wrapper(transaction_nonce.0)));
-                if transaction_nonce > sender_nonce {
-                    valid_transaction_builder = valid_transaction_builder
-                        .and_requires((sender_address, Felt252Wrapper(transaction_nonce.0 - FieldElement::ONE)));
+            match tx_priority_info {
+                // Make sure txs from same account are executed in correct order (nonce based ordering)
+                TxPriorityInfo::RegularTxs { sender_address, transaction_nonce, sender_nonce } => {
+                    valid_transaction_builder =
+                        valid_transaction_builder.and_provides((sender_address, Felt252Wrapper(transaction_nonce.0)));
+                    if transaction_nonce > sender_nonce {
+                        valid_transaction_builder = valid_transaction_builder
+                            .and_requires((sender_address, Felt252Wrapper(transaction_nonce.0 - FieldElement::ONE)));
+                    }
                 }
+                TxPriorityInfo::L1Handler { nonce } => {
+                    valid_transaction_builder =
+                        valid_transaction_builder.and_provides(("0x0", Felt252Wrapper(nonce.0)));
```

Comment:
Here you use a String as tag. I think we still want to use a `ContractAddres` (with is a `Felt252Wrapper`). 
So I would use `Felt252Wrapper::ZERO` instead